### PR TITLE
Refactoring of permissions (part 2)

### PR DESCRIPTION
**Pull request by skrchnavy**
this is follow up of #360 amd #282 to fix bugs in permissions and also to remove Organisations concept  from UV and replaced by actor concept.

### CODE REVIEW

**Comment by skrchnavy** [comment]
@eea03 please chek if #345 is also included in the fixes

**Comment by eea03** [comment]
This pull request depends on changes in https://github.com/UnifiedViews/Plugin-DevEnv/pull/45

File: frontend/src/main/webapp/WEB-INF/config.sample.properties [hunk 1]
```diff
@@ -114,18 +114,25 @@ admin.permission=administrator
 #CAS logout URL
 #cas.logout.url=https://192.168.1.8:8443/cas2/logout?service=http%3A%2F%2F192.168.7.42%3A18080%2Funifiedviews%2F
 
-#type of ownership user|organization
-ownership.type=user
+#CAS attribute representing full name
+cas.subjectIdAttributeName = Subject.UPVSIdentityID
+
+#CAS attribute representing full name
+cas.fullNameAttributeName = Subject.FormattedName
```
Comment (skrchnavy) [inline frontend/src/main/webapp/WEB-INF/config.sample.properties]: proposed change of name to cas.attributeName.fullName
also relevant for others to improve readability

File: backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/context/Context.java [hunk 2]
```diff
@@ -413,4 +413,9 @@ public Long getPipelineExecutionId() {
     public Long getDpuInstanceId() {
         return dpuInstance.getId();
     }
+
```
Comment (skrchnavy) [inline backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/context/Context.java]: move this part before getPipelineOwnerActorId() to unify diff

**Comment by eea03** [comment]
Pull request is ready to be merged;
Short sumarization of purpose and changes in this pull request:
- removing of organization concept from UV
- introducing actor concept - actor contains metadata about the user but does not influence the implementation and behavior of UV (like organization concept did)
- fixing bugs in permissions
- cleaning unused permissions, adding identified new permissions, unifying redundant permissions
- refactoring of permissions code

List of all permissions mapped to specific roles can be found [here](https://team.eea.sk/wiki/pages/worddav/preview.action?fileName=zoznam_opravneni_mapovanie_roly.xlsx&pageId=110461447)
List of all permissions with short description (work in process) can be found [here](https://team.eea.sk/wiki/display/OD/Reimplementacia+opravneni#Reimplementaciaopravneni-Popisopravneni)

Please @tomas-knap revise

**Comment by tomas-knap** [comment]
reviewing

**Comment by tomas-knap** [comment]
Please add mysql/data.sql update script, as certain permissions were updated in data.sql and it is not possible to rerun data.sql, because that would destroy certain data which are there. Maybe it is enough to have script which deletes data from permission, usr_usr_role tables and reinsert

**Comment by eea03** [comment]
Permissions updates are included in MySQL update script https://github.com/UnifiedViews/Core/blob/feature/permissionsRefactoring/db/mysql/updates/2.1.0-update.sql
Migration from 2.0.1 version was tested and compared to database created from blank installation.

However I think too that this should be separated and schema and data updates should be separate scripts. But this was beyond the scope of this pull request and this mechanism was introduced long before this pull request.
I would suggest to create a separate issue for this.

**Comment by tomas-knap** [comment]
Created issues for separation of update scripts for schema/data.sql: https://github.com/UnifiedViews/Core/issues/397

**Pull request merged**